README.md

```diff
@@ -20,3 +20,22 @@ To note that [there is already a request](https://github.com/containers/storage/
 * [Podman Github](https://github.com/containers/libpod)
 * [Podman Storage](https://github.com/containers/storage)
 * [Podman pluging for crfs](https://github.com/giuseppe/crfs-plugin)
+
+## Details and problems
+
+The first thing I ended up looking at is the [`additionalimagestores`][https://github.com/containers/storage/blob/master/docs/containers-storage.conf.5.md#storage-options-table] (From the doc: Paths to additional container image stores. Usually these are read/only and stored on remote network shares.) 
+
+This setting seems perfect for our use case, unfortunately it might not work for us.
+
+The variable it is used here: https://github.com/containers/storage/blob/abb30b354de85f9e85bb454d374c5bfebbda9d51/drivers/overlay/overlay.go#L861 and it is just pre-pended to the path of the layer we want to mount.
+
+Unfortunately the path of the layer comes in the format `overlay/l/XMXW6ERNXS2BPZXP23PXJBTWTV`, where `XMXW6ERNXS2BPZXP23PXJBTWTV` is a random number and the whole path is a symlink to the unpacked layer directory.
```

Comment:
the `overlay/l/$SHORTID` is a symlink to `overlay/$FULLNAME` that is handled internally by containers/storage.

The options in `mount(2)` are limited to 1 memory page (4096 kb).  The symlink trick is to make it harder to hit it.

Reply:
What I am still missing is what `$FULLNAME` is.

It is not the layer digest, can it be the ChainID?

Reply:
that is the layer ID, as stored in the layers store (`overlay-layers/layers.json`).

@rhatdan @nalind is there anything we could do to simplify how additional stores are generated?

Reply:
Indeed I explored that file and it fits perfectly. What I am still unsure is how the layerID is computed.

It seems that is some reproducible way, because if I erase the storage and re-download the image I get always the same IDs. But now I am not sure what schema is used to compute the layer id. I was thinking about the ChainID, but it does not compute.

```
{
    "id": "5e136cc2daffbdeb51c47fef3dc5c22d17cdf0df7d97e5a30cb5d4ff5f44eace",
    "parent": "814c70fdae62bc26c603bfae861f00fb1c77fc0b1ee8d565717846f4df24ae5d",
    "created": "2020-01-24T14:34:01.67057916Z",
    "compressed-diff-digest": "sha256:3cae3b79eb09474b8f1af0d0ba7a7eb8a00d9ae59f4db2571867d9253ff231ec",
    "compressed-size": 44897060,
    "diff-digest": "sha256:8757aa86cdf845cf2e079f296e1decb09098fe9501045f17977ac2d50beca90c",
    "diff-size": 127762944,
    "compression": 2,
    "uidset": [
      0,
      100,
      101,
      103
    ],
    "gidset": [
      0,
      4,
      42,
      43,
      50,
      101
    ]
  },


$ echo -n "814c70fdae62bc26c603bfae861f00fb1c77fc0b1ee8d565717846f4df24ae5d 8757aa86cdf845cf2e079f296e1decb09098fe9501045f17977ac2d50beca90c" | sha256sum 
97289b5e6e416ac3ceb32f09e651278160de6289bc6390faaa94d7f1b7e505b4  -
```

Maybe there is some error in my computation of the ChainID. Or maybe it is something else...

Reply:
@mtrmac @nalind can  you answer the questions above?

Reply:
Got it!

I believe it is a minor difference on how the layer id is computed.

[Here we have:](https://github.com/containers/image/blob/54f8af26146407f9e7e7be665e0a10d1476ce85e/storage/storage_image.go#L678)
```
id = digest.Canonical.FromBytes([]byte(lastLayer + "+" + diffID.Hex())).Hex()
```
while I was expecting it to be
```
id = digest.Canonical.FromBytes([]byte(lastLayer + " " + diffID.Hex())).Hex()
```

Note the ` ` (space) instead of the `+` character.

The second version is what is usually called `chainID`, while the first one is not commonly used, as far as I know.

However, there may be good reasons why this other merge function was choose...